src/dataverk/datapackage.py

```diff
@@ -91,102 +97,41 @@ def path(self):
         :return: None
         """
-        self._verify_add_resource_input_types(df, resource_name, resource_description)
-        resource_name = file_functions.remove_whitespace(resource_name)
-        self.resources[resource_name] = self._get_schema(df=df, path=self.path, resource_name=resource_name,
-                                                         resource_description=resource_description, format=format,
-                                                         compress=compress, dsv_separator=dsv_separator, spec=spec)
-        self.resources[resource_name]['df'] = df
-        self._datapackage_metadata["datasets"][resource_name] = resource_description
-        self._datapackage_metadata['resources'].append(self._get_schema(df=df, path=self.path,
-                                                                        resource_name=resource_name,
-                                                                        resource_description=resource_description,
-                                                                        format=format, compress=compress,
-                                                                        dsv_separator=dsv_separator, spec=spec))
-
-    def add_remote_resource(self, resource_url: str, resource_description: str=""):
-        resource_name, resource_fmt = self._resource_name_and_type_from_url(resource_url)
-        self._datapackage_metadata['datasets'][resource_name] = resource_description
-        self._datapackage_metadata['resources'].append({
-            'name': resource_name,
-            'description': resource_description,
-            'path': resource_url,
-            'format': resource_fmt
-        })
 
-    @staticmethod
-    def _resource_name_and_type_from_url(resource_url):
-        parsed_url = url.parse_url(resource_url)
+        formatted_resource = get_resource_object(resource_type=resource_type, resource=resource,
+                                                 datapackage_path=self.path, resource_name=resource_name,
+                                                 resource_description=resource_description, spec=spec).get_schema()
+
+        formatted_resource_name = formatted_resource.get('name')
+
+        self.resources[formatted_resource_name] = formatted_resource
 
-        if not parsed_url.scheme == "https" and not parsed_url.scheme == "http":
-            raise ValueError(f"Remote resource needs to be a web address, scheme is {parsed_url.scheme}")
+        if resource_type == 'df':
+            self.datapackage_metadata['datasets'][formatted_resource_name] = resource_description
+            self.resources[formatted_resource_name]['df'] = resource
 
-        resource = parsed_url.path.split('/')[-1]
-        resource_name_and_format = resource.split('.', 1)
-        return resource_name_and_format[0], resource_name_and_format[1]
+        self._datapackage_metadata['resources'].append(formatted_resource)
 
+    # TODO: IMPLEMENT THIS METHOD
```

Comment:
Jeg er ikke enig i det. For min del burde denne metoden vært fjernet for lengst (vi bruker den ikke allikevel).

- Hva er hensikten med å typesjekke alle input parametere til metodene våre? Det eneste jeg ser av fordeler med dette er at vi kan kaste en TypeError raskt og kanskje spare brukeren for noe scrolling i stacktracen, men er det verdt merarbeidet og ikke minst vedlikeholdskravet? Det vil jo kræsje allikevel når noe går galt, og klarer vi ikke gi en mer enn tilstrekkelig beskrivelse av apiene våre ved bruk av typehinting og docstrings?
- Er tanken at vi skal legge på typesjekk av inputparametere på alle metoder og funksjoner i dataverk? Hvis nei, hvorfor er add_resource() spesiell? Hvis ja, så har jeg vanskelig for å se hvordan vi skal kunne abstrahere noe som helst, som i prinsippet vil si at vi må lage to metoder for hver metode vi lager: metoden og en "typevalideringsfunksjon" for metoden.
- Hva hvis inputparameteren kan være flere forskjellige ting slik som f.eks. resource i add_resource() over (DataFrame, bytes, str), skal vi da isinstance sjekke mot en liste med "godkjente" typer?

Jeg mener heller typehinting og godt beskrivende docstrings er veien å gå for å gjøre biblioteket enkelt og forståelig for brukeren, og klarer ikke å se noen særlig tungtveiende argumenter for å legge på disse typesjekkene, i hvert fall ikke argumenter som utjevner ulempene.

Jeg mener også at add_resource() burde kjøre en distribution validation (fra dcat-validator pakken) på metadataen til ressursen som legges til og så presentere en rapport ala den som nå presenteres når Datapackage objektet instansieres. Dette vil være verdifullt fordi vi da kan sjekke at metadataen samsvarer med dcat-standarden før vi poster det til elastic search indexen.

Reply:
Jeg klarer ikke å finne noen argumenter for at vi skal beholde det, så enig. Fikset